### PR TITLE
Add more set ops, documentation, tests

### PR DESCRIPTION
**Pull request by jameysharp**
I just found this crate and I love not having to write these functions again in each project myself. Thank you for releasing it!

This pull request has an assortment of (what I think are) improvements I'd like to propose. I only set out to add the `is_disjoint`, `is_subset`, `is_superset` methods, but noticed various other things along the way. Each is pretty small, so although I made them separate commits, I didn't feel like making separate pull requests. If you don't want some of them I can rebase away those commits if you'd like.

### CODE REVIEW

**Comment by aj-alonso** [comment]
These methods are very useful. Is there any issue preventing them from being merged? I can help.

**Comment by rklaehn** [comment]
Hey, sorry about not seeing this earlier. Looks good.

So this

- adds is_disjoint, is_subset, is_superset
- implements size_hint
- uses checked addition and subtraction

correct?

I think it is fine. @aj-alonso maybe you can take another look. Otherwise I will merge this and publish a new release tomorrow.

File: src/lib.rs [hunk 1]
```diff
@@ -158,50 +160,110 @@ use crate::sorted_pair_iterator::*;
 
 /// set operations for iterators where the items are sorted according to the natural order
 pub trait SortedIterator: Iterator + Sized {
-    /// union with another sorted iterator
-    fn union<J>(self, that: J) -> Union<Self, J>
+    /// Visits the values representing the union, i.e., all the values in `self` or `other`,
+    /// without duplicates.
+    fn union<J>(self, other: J) -> Union<Self, J>
```
Comment (rklaehn) [inline src/lib.rs:165]: I use this/that from back when I did java/scala. Now this is self, but old habits die hard.

But I guess other works as well :-)

File: src/sorted_iterator.rs [hunk 2]
```diff
@@ -173,13 +173,16 @@ where
...
+                max(lo, ilo),
+                hi.and_then(|hi| ihi.and_then(|ihi| hi.checked_add(ihi))),
+            )
         })
     }
 }
 
 pub struct Intersection<I: Iterator, J: Iterator> {
-    pub(crate) a: Peekable<I>,
+    pub(crate) a: I,
```
Comment (rklaehn) [inline src/sorted_iterator.rs:185]: Ah, so the optimization is that one side does not have to be wrapped in peekable. Nice.

**Review by rklaehn — APPROVED** [review]

**Comment by rklaehn** [comment]
@jameysharp , really sorry that I did not see this before. I had quite a bit of stuff going on this year - bought a house and moved from Germany to Romania with the entire family... Kinda distracting...

**Comment by jameysharp** [comment]
@rklaehn, I completely understand! I figured it was something like that. Anyway, I appreciate you taking a look at this now. And I appreciate you too, @aj-alonso, for calling attention to these patches. :grin:

**Comment by aj-alonso** [comment]
Excellent! Looks good to me too. I understand that life sometimes gets in the way :-) Thank you for your work on this beatiful library @rklaehn and for your patches @jameysharp !

**Comment by rklaehn** [comment]
Published 0.1.8

**Pull request merged**